SingularityService/src/main/java/com/hubspot/singularity/auth/SingularityGroupsScopesAuthorizer.java

```diff
@@ -239,52 +185,41 @@ private boolean isJita(SingularityUser user) {
   }
 
   private boolean isAdmin(SingularityUser user) {
-    return groupsIntersect(scopesConfiguration.getAdmin(), user.getScopes());
+    return hasScope(user, SingularityAuthorizationScope.ADMIN);
   }
 
-  private boolean hasReadScope(SingularityUser user) {
-    return (
-      groupsIntersect(scopesConfiguration.getRead(), user.getScopes()) ||
-      groupsIntersect(scopesConfiguration.getWrite(), user.getScopes())
-    );
-  }
-
-  private void checkReadScope(SingularityUser user) {
-    checkForbidden(
-      hasReadScope(user),
-      "%s must have one or more scopes to READ: %s, %s",
-      user.getId(),
-      scopesConfiguration.getRead(),
-      scopesConfiguration.getWrite()
-    );
-  }
-
-  private boolean hasWriteScope(SingularityUser user) {
-    return groupsIntersect(scopesConfiguration.getWrite(), user.getScopes());
+  private Set<String> getScopes(SingularityAuthorizationScope scope) {
+    switch (scope) {
+      case READ:
+        return Sets.union(scopesConfiguration.getRead(), scopesConfiguration.getWrite());
+      case WRITE:
+        return scopesConfiguration.getWrite();
+      case DEPLOY:
```

Comment:
If it makes it simpler in the overall implementation, I'm fine with having the DEPLOY scope behave similar to the new ones (i.e. falls under WRITE unless otherwise overriden by a request). I had originally separated out DEPLOY for future use with our internal automation, but it isn't turned on or enforced by default

Reply:
Do you mean change DEPLOY to not fall back to use WRITE? I set up EXEC so that it doesn't fall back to write, but that the request group gets EXEC privs unless explicitly removed by the overrides.

Reply:
We're going to have to add all these scopes for all the appropriate people anyways, it would be easy to add DEPLOY for them too